PR#5098 – Jekyll Dependency Error

- [ ] Dependency Error: Yikes! It looks like you don't have jekyll_ai_related_posts or one of its dependencies installed. In order to use Jekyll as currently configured, you'll need to install this gem. If you've run Jekyll with `bundle exec`, ensure that you have included the jekyll_ai_related_posts gem in your Gemfile as well. The full error message from Ruby is: 'cannot load such file -- jekyll_ai_related_posts' If you run into trouble, you can find helpful resources at https://jekyllrb.com/help/! 

```
Run bundle exec jekyll build --trace --incremental --baseurl "/RalphHightower"
  bundle exec jekyll build --trace --incremental --baseurl "/RalphHightower"
  shell: /usr/bin/bash -e {0}
  env:
    GITHUB_PAGES: true
    JEKYLL_ENV: production
    JEKYLL_GITHUB_TOKEN: ***
    LOG_LEVEL: debug
To use retry middleware with Faraday v2.0+, install `faraday-retry` gem
Configuration file: /home/runner/work/RalphHightower/RalphHightower/_config.yml
  Dependency Error: Yikes! It looks like you don't have jekyll_ai_related_posts or one of its dependencies installed. In order to use Jekyll as currently configured, you'll need to install this gem. If you've run Jekyll with `bundle exec`, ensure that you have included the jekyll_ai_related_posts gem in your Gemfile as well. The full error message from Ruby is: 'cannot load such file -- jekyll_ai_related_posts' If you run into trouble, you can find helpful resources at https://jekyllrb.com/help/! 
bundler: failed to load command: jekyll (/home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/bin/jekyll)
/home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/external.rb:70:in 'block in Jekyll::External.require_with_graceful_fail': jekyll_ai_related_posts (Jekyll::Errors::MissingDependencyException)
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/external.rb:55:in 'Array#each'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/external.rb:55:in 'Jekyll::External.require_with_graceful_fail'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/plugin_manager.rb:30:in 'Jekyll::PluginManager#require_gems'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/plugin_manager.rb:22:in 'Jekyll::PluginManager#conscientious_require'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/site.rb:131:in 'Jekyll::Site#setup'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/site.rb:36:in 'Jekyll::Site#initialize'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/commands/build.rb:30:in 'Class#new'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/commands/build.rb:30:in 'Jekyll::Commands::Build.process'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/command.rb:91:in 'block in Jekyll::Command.process_with_graceful_fail'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/command.rb:91:in 'Array#each'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/command.rb:91:in 'Jekyll::Command.process_with_graceful_fail'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/commands/build.rb:18:in 'block (2 levels) in Jekyll::Commands::Build.init_with_program'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in 'block in Mercenary::Command#execute'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in 'Array#each'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in 'Mercenary::Command#execute'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/mercenary-0.4.0/lib/mercenary/program.rb:44:in 'Mercenary::Program#go'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/mercenary-0.4.0/lib/mercenary.rb:21:in 'Mercenary.program'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/exe/jekyll:15:in '<top (required)>'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/bin/jekyll:25:in 'Kernel#load'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/bin/jekyll:25:in '<top (required)>'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/3.4.0/bundler/cli/exec.rb:59:in 'Kernel.load'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/3.4.0/bundler/cli/exec.rb:59:in 'Bundler::CLI::Exec#kernel_load'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/3.4.0/bundler/cli/exec.rb:23:in 'Bundler::CLI::Exec#run'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/3.4.0/bundler/cli.rb:452:in 'Bundler::CLI#exec'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/3.4.0/bundler/vendor/thor/lib/thor/command.rb:28:in 'Bundler::Thor::Command#run'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/3.4.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in 'Bundler::Thor::Invocation#invoke_command'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/3.4.0/bundler/vendor/thor/lib/thor.rb:538:in 'Bundler::Thor.dispatch'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/3.4.0/bundler/cli.rb:35:in 'Bundler::CLI.dispatch'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/3.4.0/bundler/vendor/thor/lib/thor/base.rb:584:in 'Bundler::Thor::Base::ClassMethods#start'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/3.4.0/bundler/cli.rb:29:in 'Bundler::CLI.start'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/gems/3.4.0/gems/bundler-2.6.9/exe/bundle:28:in 'block in <top (required)>'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/3.4.0/bundler/friendly_errors.rb:117:in 'Bundler.with_friendly_errors'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/gems/3.4.0/gems/bundler-2.6.9/exe/bundle:20:in '<top (required)>'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/bin/bundle:25:in 'Kernel#load'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/bin/bundle:25:in '<main>'
/opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/3.4.0/bundled_gems.rb:82:in 'Kernel.require': cannot load such file -- jekyll_ai_related_posts (LoadError)
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/3.4.0/bundled_gems.rb:82:in 'block (2 levels) in Kernel#replace_require'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/external.rb:57:in 'block in Jekyll::External.require_with_graceful_fail'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/external.rb:55:in 'Array#each'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/external.rb:55:in 'Jekyll::External.require_with_graceful_fail'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/plugin_manager.rb:30:in 'Jekyll::PluginManager#require_gems'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/plugin_manager.rb:22:in 'Jekyll::PluginManager#conscientious_require'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/site.rb:131:in 'Jekyll::Site#setup'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/site.rb:36:in 'Jekyll::Site#initialize'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/commands/build.rb:30:in 'Class#new'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/commands/build.rb:30:in 'Jekyll::Commands::Build.process'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/command.rb:91:in 'block in Jekyll::Command.process_with_graceful_fail'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/command.rb:91:in 'Array#each'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/command.rb:91:in 'Jekyll::Command.process_with_graceful_fail'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/lib/jekyll/commands/build.rb:18:in 'block (2 levels) in Jekyll::Commands::Build.init_with_program'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in 'block in Mercenary::Command#execute'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in 'Array#each'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in 'Mercenary::Command#execute'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/mercenary-0.4.0/lib/mercenary/program.rb:44:in 'Mercenary::Program#go'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/mercenary-0.4.0/lib/mercenary.rb:21:in 'Mercenary.program'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/gems/jekyll-4.4.1/exe/jekyll:15:in '<top (required)>'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/bin/jekyll:25:in 'Kernel#load'
	from /home/runner/work/RalphHightower/RalphHightower/vendor/bundle/ruby/3.4.0/bin/jekyll:25:in '<top (required)>'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/3.4.0/bundler/cli/exec.rb:59:in 'Kernel.load'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/3.4.0/bundler/cli/exec.rb:59:in 'Bundler::CLI::Exec#kernel_load'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/3.4.0/bundler/cli/exec.rb:23:in 'Bundler::CLI::Exec#run'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/3.4.0/bundler/cli.rb:452:in 'Bundler::CLI#exec'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/3.4.0/bundler/vendor/thor/lib/thor/command.rb:28:in 'Bundler::Thor::Command#run'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/3.4.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in 'Bundler::Thor::Invocation#invoke_command'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/3.4.0/bundler/vendor/thor/lib/thor.rb:538:in 'Bundler::Thor.dispatch'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/3.4.0/bundler/cli.rb:35:in 'Bundler::CLI.dispatch'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/3.4.0/bundler/vendor/thor/lib/thor/base.rb:584:in 'Bundler::Thor::Base::ClassMethods#start'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/3.4.0/bundler/cli.rb:29:in 'Bundler::CLI.start'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/gems/3.4.0/gems/bundler-2.6.9/exe/bundle:28:in 'block in <top (required)>'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/3.4.0/bundler/friendly_errors.rb:117:in 'Bundler.with_friendly_errors'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/lib/ruby/gems/3.4.0/gems/bundler-2.6.9/exe/bundle:20:in '<top (required)>'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/bin/bundle:25:in 'Kernel#load'
	from /opt/hostedtoolcache/Ruby/3.4.7/x64/bin/bundle:25:in '<main>'
Error: Process completed with exit code 1.
0s
```

_Originally posted by @RalphHightower in https://github.com/RalphHightower/RalphHightower/issues/5098#issuecomment-3941102509_

--- a/Gemfile
+++ b/Gemfile
@@ -18,6 +18,7 @@ gem 'minima', '~> 2.5', '>= 2.5.2'
 
 # If you have any plugins, put them here!
 group :jekyll_plugins do
+gem 'jekyll_ai_related_posts', '~> 0.1.4'
 gem 'jekyll-avatar', '~> 0.8.0'
 gem 'jekyll-default-layout', '~> 0.1.5'
 gem 'jekyll-feed', '~> 0.17.0'

